### PR TITLE
Secrets using bind mounts

### PR DESCRIPTION
**Pull request by dnephin**
Fake secrets in `docker-compose` using bind mounts

See https://github.com/docker/docker/pull/30144 for the feature in `docker stack deploy`

### CODE REVIEW

**Comment by shin-** [comment]
@dnephin Do you need help looking into those?

```
20:17:21 [all_py27] =================================== FAILURES ===================================

20:17:21 [all_py27] ________________ ProjectTest.test_project_up_healthy_dependency ________________

20:17:21 [all_py27] tests/integration/testcases.py:59: in wrapper

20:17:21 [all_py27]     return f(self, *args, **kwargs)

20:17:21 [all_py27] tests/integration/project_test.py:1431: in test_project_up_healthy_dependency

20:17:21 [all_py27]     config_data = build_config(config_dict)

20:17:21 [all_py27] E   TypeError: build_config() takes exactly 0 arguments (1 given)

20:17:21 [all_py27] ____________ ProjectTest.test_project_up_no_healthcheck_dependency _____________

20:17:21 [all_py27] tests/integration/testcases.py:59: in wrapper

20:17:21 [all_py27]     return f(self, *args, **kwargs)

20:17:21 [all_py27] tests/integration/project_test.py:1504: in test_project_up_no_healthcheck_dependency

20:17:21 [all_py27]     config_data = build_config(config_dict)

20:17:21 [all_py27] E   TypeError: build_config() takes exactly 0 arguments (1 given)

20:17:21 [all_py27] _______________ ProjectTest.test_project_up_unhealthy_dependency _______________

20:17:21 [all_py27] tests/integration/testcases.py:59: in wrapper

20:17:21 [all_py27]     return f(self, *args, **kwargs)

20:17:21 [all_py27] tests/integration/project_test.py:1468: in test_project_up_unhealthy_dependency

20:17:21 [all_py27]     config_data = build_config(config_dict)

20:17:21 [all_py27] E   TypeError: build_config() takes exactly 0 arguments (1 given)

20:17:21 [all_py27] ___________________ ProjectTest.test_project_up_with_secrets ___________________

20:17:21 [all_py27] tests/integration/testcases.py:59: in wrapper

20:17:21 [all_py27]     return f(self, *args, **kwargs)

20:17:21 [all_py27] tests/integration/project_test.py:1184: in test_project_up_with_secrets

20:17:21 [all_py27]     assert output == "This is the secret\n"

20:17:21 [all_py27] E   AssertionError: assert 'cat: read er...a directory\n' == 'This is the secret\n'

20:17:21 [all_py27] E     - cat: read error: Is a directory

20:17:21 [all_py27] E     + This is the secret
```

**Comment by dnephin** [comment]
Oops, I didn't notice. I'll look into them

**Comment by dnephin** [comment]
I fixed the TypeErrors, but the other test is passing locally.

From the failure it sounds like the file doesn't exist on the "host", so the bind mount is creating the path as a directory.

**Comment by dnephin** [comment]
ok, tests are fixed

File: tests/integration/project_test.py [hunk 1]
```diff
@@ -1515,3 +1517,31 @@ def test_project_up_no_healthcheck_dependency(self):
...
             svc1.is_healthy()
+
+
+def create_host_file(client, filename):
+    dirname = os.path.dirname(filename)
+
+    with open(filename, 'r') as fh:
+        content = fh.read()
+
+    print(content)
```
Comment (shin-) [inline tests/integration/project_test.py]: Remove debug statement?

Comment (dnephin) [inline tests/integration/project_test.py]: oops, removed

**Pull request merged**